src/libraries/Common/src/System/Net/SocketProtocolSupportPal.cs

```diff
@@ -12,7 +13,8 @@ internal static partial class SocketProtocolSupportPal
 
         public static bool OSSupportsIPv6 { get; } = IsSupported(AddressFamily.InterNetworkV6) && !IsIPv6Disabled();
         public static bool OSSupportsIPv4 { get; } = IsSupported(AddressFamily.InterNetwork);
-        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix);
+        // iOS/tvOS ostensibly has AF_UNIX, but throws EPERM on iOS/tvOS 10.0+ on bind
+        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix) && !RuntimeInformation.IsOSPlatform(OSPlatform.Create("TVOS")) && !RuntimeInformation.IsOSPlatform(OSPlatform.Create("IOS"));
```

Comment:
```suggestion
        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix) && !OperatingSystem.IsIOS() && !OperatingSystem.IsTvOS();
```
These are the right API to use to check for OS. `RuntimeInformation.IsOSPlatform` is deprecated.